api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/ApplicationConfigurationOntologyUtils.java

```diff
@@ -18,112 +21,228 @@
+            			log.warn("Couldn't look up the faux property", e);
+            		}
+                }
+            }
+        } finally {
+            qe.close();
         }
-
-        return additionalProps;
+        return fauxDataProps;
     }
 
-    private static boolean appropriateDomain(Resource domainRes, Individual subject, Model tboxModel) {
-        if (subject == null || domainRes == null) {
+	private static Query createQuery(Property op, boolean optionalRange) {
+		String queryStr = getFauxPropQuery(op.getURI(), optionalRange);
+        log.debug(queryStr);
+        Query q = QueryFactory.create(queryStr);
+		return q;
+	}
+
+    private static boolean isDomainMatchSubject(String domainUri, Individual subject) {
+        if (subject == null || domainUri == null) {
             return true;
         }
-        for (VClass vclass : subject.getVClasses()) {
-            if ((vclass.getURI() != null) &&
-                    ((vclass.getURI().equals(domainRes.getURI()) ||
-                    (tboxModel.contains(
-                            ResourceFactory.createResource(
-                                    vclass.getURI()), RDFS.subClassOf, domainRes))))) {
+        Set<String> vClassUris = subject.getVClasses().stream().map(vclass -> vclass.getURI()).collect(Collectors.toSet());
+		for (String vClassUri : vClassUris) {
+            if (vClassUri != null && vClassUri.equals(domainUri)) {
                 return true;
             }
         }
         return false;
     }
 
+	public static List<Property> getPossibleFauxProps(List<? extends FauxPropertyWrapper> curProps, Individual subject,	VitroRequest vreq, boolean isData) {
+		Model displayModel = vreq.getDisplayModel();
+        Model tboxModel = vreq.getOntModelSelector().getTBoxModel();
+		Model union = ModelFactory.createUnion(displayModel, tboxModel);
+		Map<String, FauxProperty> curPropsMap = curProps.stream().collect(Collectors.toMap(FauxPropertyWrapper::getContextUri, FauxPropertyWrapper::getFauxProperty));
+		WebappDaoFactory wadf = new WebappDaoFactoryJena(ModelFactory.createOntologyModel(OntModelSpec.OWL_MEM, union));
+		FauxPropertyDao fpDao = wadf.getFauxPropertyDao();
+		ObjectPropertyDao opDao = wadf.getObjectPropertyDao();
+		DataPropertyDao dpDao = wadf.getDataPropertyDao();
+        
+	    Query q = QueryFactory.create(getPossibleFauxQuery(isData));
+        QueryExecution qe = QueryExecutionFactory.create(q, union);
+	    List<Property> fauxDataProps = new ArrayList<Property>();
+        try {
+            ResultSet rs = qe.execSelect();
+            while (rs.hasNext()) {
+                QuerySolution qsoln = rs.nextSolution();
+                Resource domainRes = qsoln.getResource("domain");
+                String domainURI = (domainRes != null) ? domainRes.getURI() : null;
+                //Resource rangeRes = qsoln.getResource("range");
+                //String rangeURI = (rangeRes != null) ? rangeRes.getURI() : null;
```

Comment:
similar question, why we should keep this in comment?

Reply:
Removed